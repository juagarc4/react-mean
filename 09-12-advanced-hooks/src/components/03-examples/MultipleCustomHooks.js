import React from 'react'
import { useFetch } from 'hooks/useFetch'
import { useCounter } from 'hooks/useCounter'

export const MultipleCustomHooks = () => {
  const { counter, increment, decrement } = useCounter()

  const url = `https://www.breakingbadapi.com/api/quotes/${counter}`
  const { loading, data } = useFetch(url)
  const { author, quote } = !!data && data[0]
  const id = !!data && data[0].quote_id
  return (
    <div>
      <h1>Breaking Bad quotes</h1>
      <hr />
      {loading ? (
        <div className="alert alert-info text-center">Loading...</div>
      ) : (
        <blockquote className="blockquote text-justify">
          <p>
            {id} - {quote}
          </p>
          <footer className="blockquote-footer">{author}</footer>
        </blockquote>
      )}
      {counter > 1 && (
        <button className="btn btn-primary" onClick={decrement}>
          Previous quote
        </button>
      )}
      <button className="btn btn-primary" onClick={increment}>
        Next quote
      </button>
    </div>
  )
}
