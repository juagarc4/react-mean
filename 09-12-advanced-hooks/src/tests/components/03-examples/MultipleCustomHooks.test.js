import React from 'react'
import { shallow } from 'enzyme'
import { MultipleCustomHooks } from 'components/03-examples/MultipleCustomHooks'
import { useFetch } from 'hooks/useFetch'
import { useCounter } from 'hooks/useCounter'
jest.mock('hooks/useFetch')
jest.mock('hooks/useCounter')

describe('Tests in <MultipleCustomHooks />', () => {
  test('Should show correctly', () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    })
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => {},
    })

    const wrapper = shallow(<MultipleCustomHooks />)
    expect(wrapper).toMatchSnapshot()
  })

  test('Should show the info', () => {
    useFetch.mockReturnValue({
      data: [
        {
          quote_id: 1,
          author: 'Raul',
          quote: 'Hello world',
        },
      ],
      loading: false,
      error: null,
    })
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => {},
    })
    const wrapper = shallow(<MultipleCustomHooks />)
    expect(wrapper.find('.alert').exists()).toBe(false)
    expect(wrapper.find('.blockquote p').text().trim()).toBe('1 - Hello world')
    expect(wrapper.find('footer').text().trim()).toBe('Raul')
  })
})
