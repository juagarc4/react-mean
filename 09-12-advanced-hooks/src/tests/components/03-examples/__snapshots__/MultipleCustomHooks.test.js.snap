// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <MultipleCustomHooks /> Should show correctly 1`] = `
<div>
  <h1>
    Breaking Bad quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
  >
    Previous quote
  </button>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</div>
`;
